pkg/addonmanager/controllers/addoninstall/controller.go

```diff
@@ -107,19 +107,49 @@ func (c *addonInstallController) applyAddon(ctx context.Context, addonName, clus
 				InstallNamespace: installNamespace,
 			},
 		}
-		_, err = c.addonClient.AddonV1alpha1().ManagedClusterAddOns(clusterName).Create(ctx, addon, metav1.CreateOptions{})
+		_, err := c.addonClient.AddonV1alpha1().ManagedClusterAddOns(clusterName).
+			Create(ctx, addon, metav1.CreateOptions{})
 		return err
 	case err != nil:
 		return err
 	}
 
-	if addon.Spec.InstallNamespace == installNamespace {
+	// reconcile install namespace
+	if addon.Spec.InstallNamespace != installNamespace {
+		addon = addon.DeepCopy()
+		addon.Spec.InstallNamespace = installNamespace
+		addon, err = c.addonClient.AddonV1alpha1().ManagedClusterAddOns(clusterName).
+			Update(ctx, addon, metav1.UpdateOptions{})
+		if err != nil {
+			return err
+		}
 		return nil
 	}
 
-	addon = addon.DeepCopy()
-	addon.Spec.InstallNamespace = installNamespace
-	_, err = c.addonClient.AddonV1alpha1().ManagedClusterAddOns(clusterName).Update(ctx, addon, metav1.UpdateOptions{})
+	// reconcile health check mode
```

Comment:
I think the status tracking should be independent with install addon. If an addon is not auto installed, it should still be able to define a health checker to update status. It might be put in a separated controller.

Reply:
separated an addon health check controller